client/src/pages/DetailPage.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect, useCallback, useContext } from 'react';
+import styled from '@emotion/styled';
+import Header from '../components/Header/Header';
+import Title from '../components/Header/Title';
+import Action from '../components/Header/Action';
+import Label from '../components/Header/Label';
+import { Arrow, Delete } from '../assets/icons/Actions';
+import { NavLink, useParams } from 'react-router-dom';
+import MainArea from '../components/MainArea';
+import CardDetail from '../components/CardDetail';
+import useHttp from '../hooks/useHttp.hook';
+import AuthContext from '../context/AuthContext';
+
+const Container = styled.div`
+  display: flex;
+  flex-flow: column nowrap;
+`;
+
+const TitleCenter = styled(Title)`
+  margin: 30px 0 0 0;
+  text-align: center;
+`;
+
+const Link = styled(NavLink)`
+  position: absolute;
+  z-index: 1;
+  text-decoration: none;
+`;
+const ActionRight = styled(NavLink)`
+  position: absolute;
+  z-index: 1;
+  right: 0;
+  text-decoration: none;
+`;
+const DetailPage = () => {
+  const auth = useContext(AuthContext);
+  const [setDetails, setSetDetails] = useState([]);
+  const { request } = useHttp();
+  const { setId } = useParams();
+
+  const handleDelete = async () => {
+    try {
+      await request(`/api/collection/mysets/${setId}`, 'DELETE', null, {
+        Authorization: `Bearer ${auth.token}`,
+      });
+    } catch (error) {
+      console.error(error);
+    }
```

Comment:
It looks like `useHttp` returns `request`, `loading` and `error`.
So it shouldn't be required to have additional try-catch? 
I guess this happens multiple times.

Maybe you have to take a look at your hook again because it rethrows the error in:
https://github.com/berlinum/bricks/blob/master/client/src/hooks/useHttp.hook.js#L29

In addition, you could profit by an extendable `useHttp` hook if you move the parameters to this:
```
const { request, loading, error } = useHttp(`/api/collection/mysets/${setId}`, 'DELETE', null, {
    Authorization: `Bearer ${auth.token}`,
  });
```
Then, you can call the effect from inside the hook. In this case, you can remove the other effects in this (and other) files.

Reply:
done!